drivers/network/netio/netio.c

```diff
@@ -0,0 +1,1453 @@
+        if (!NT_SUCCESS(status))
+        {
+            DPRINT1("KeWaitForSingleObject failed with status 0x%08x!\n", status);
+        }
+        ObDereferenceObject(s->ListenThread);
+        s->ListenThread = NULL;
+        s->ListenThreadHandle = NULL;
+    }
+    if (s->ListenIrp != NULL)
+    {
+        s->ListenCancelled = TRUE;
+        IoCancelIrp(s->ListenIrp);
+        s->ListenIrp = NULL;
+    }
+    if (s->ConnectionFile != NULL)
+    {
+        if (s->ConnectionFileAssociated)
+        {
+               /* This fails with error 0xc000023b (still connected) on Windows 2003 */
+            status = TdiDisassociateAddressFile(s->ConnectionFile);
+            if (!NT_SUCCESS(status))
+            {
+                DPRINT1("Warning: TdiDisassociateAddressFile returned status %08x\n", status);
+            }
+            s->ConnectionFileAssociated = FALSE;
+        }
+        ObDereferenceObject(s->ConnectionFile);
+        s->ConnectionFile = NULL;
+    }
+    if (s->ConnectionHandle != NULL)
+    {
+        ZwClose(s->ConnectionHandle);
+        s->ConnectionHandle = NULL;
+    }
+}
+
+static PWSK_SOCKET_INTERNAL SocketsToPut = NULL;
+static KSPIN_LOCK SocketsToPutListLock = 0;
+static BOOLEAN PutSocketsThreadShouldRun = FALSE;
+static KEVENT PutSocketsEvent;
+static HANDLE PutSocketsThreadHandle;
+
+static void WSKAPI PutSocketsThread(_In_opt_ void *p)
+{
+    NTSTATUS status;
+    PWSK_SOCKET_INTERNAL SocketToPut;
+    KIRQL flags;
+    int NumSocketPuts;
+
+    FUNCTION_TRACE;
+
+    while (PutSocketsThreadShouldRun)
+    {
+        status = KeWaitForSingleObject(&PutSocketsEvent, Executive, KernelMode, FALSE, NULL);
+        if (!NT_SUCCESS(status))
+        {
+            DPRINT1("KeWaitForSingleObject failed with status 0x%08x!\n", status);
+        }
+
+        KeAcquireSpinLock(&SocketsToPutListLock, &flags);
+        while (SocketsToPut != NULL)
+        {
+            SocketToPut = SocketsToPut;
+            SocketsToPut = SocketToPut->NextSocketToPut;
+            NumSocketPuts = SocketToPut->NumSocketPuts;
+
+            while (NumSocketPuts > 0)
+            {
+                SocketToPut->NumSocketPuts--;
+                NumSocketPuts = SocketToPut->NumSocketPuts;
+
+                KeReleaseSpinLock(&SocketsToPutListLock, flags);
+                SocketPut(SocketToPut);	/* Here IRQL MUST be PASSIVE_LEVEL, else we loop forever! */
+                KeAcquireSpinLock(&SocketsToPutListLock, &flags);
+            }
+        }
+        KeReleaseSpinLock(&SocketsToPutListLock, flags);
+    }
+}
+
+static void StartSocketPutThread(void)
+{
+    NTSTATUS status;
+
+    KeInitializeSpinLock(&SocketsToPutListLock);
+    KeInitializeEvent(&PutSocketsEvent, SynchronizationEvent, FALSE);
+    PutSocketsThreadShouldRun = TRUE;
+
+    status = PsCreateSystemThread(&PutSocketsThreadHandle, THREAD_ALL_ACCESS, NULL, NULL, NULL, PutSocketsThread, NULL);
+    if (status != STATUS_SUCCESS)
```

Comment:
The `NT_SUCCESS` macro is the correct syntax for this operation. This also applies to a few other places:
```suggestion
    if (!NT_SUCCESS(status))
```
https://learn.microsoft.com/en-us/windows-hardware/drivers/kernel/using-ntstatus-values

Reply:
Ok thanks applied (also to all other occurences)